var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -126,10 +126,12 @@ def edit(self, spec, prefix):
         # C++ compilers are being used to build the ESMF library.
         if self.compiler.name == 'gcc':
             os.environ['ESMF_COMPILER'] = 'gfortran'
+            gfortran_major_version = int(spack.compiler.get_compiler_version_output(self.compiler.fc, '-dumpversion').split('.')[0])
```

Comment:
I feel like there has to be an easier way to get the compiler version than this. Does `self.compiler.version.up_to(1)` work?

Reply:
Thanks for taking the time to review the PR.

I can look into this, I had asked a question in the discussions but didn't get any response: https://github.com/spack/spack/discussions/28146

Ok, I can take care of ESMF for the moment, and add other maintainers later as required.

Reply:
Ah, yeah, that might be necessary for mixed compiler environments like Clang + GFortran. Let's leave that as is for now. We're planning a big overhaul that will make compilers formal dependencies and that should make it easier to express and query this information.

Thanks for volunteering to be a maintainer!